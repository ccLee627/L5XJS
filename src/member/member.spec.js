const Member = require("./index");

describe("Member Class", () => {
    describe("New Instance", () => {
        it("Accepts Proper Input", () => {
            const fn = (
                name,
                datatype,
                desc = null,
                hidden = false,
                target = null,
                bit = null
            ) => () => new Member(name, datatype, desc, hidden, target, bit);

            expect(fn("tagName")).toThrow();
            expect(fn(null, "DINT")).toThrow();
            expect(fn("tagName", "DINT")).not.toThrow();
            expect(fn("tagName", "DINT", 12)).toThrow();
            expect(fn("tagName", "DINT", "A Description")).not.toThrow();
            expect(fn("tagName", "DINT", "A Description", 12)).toThrow();
            expect(fn("tagName", "DINT", "A Description", true)).not.toThrow();
            expect(fn("tagName", "DINT", "A Description", true, 12)).toThrow();
            expect(fn("tagName", "DINT", "A Description", true, "aTarget")).not.toThrow();
            expect(
                fn("tagName", "DINT", "A Description", true, "aTarget", "notANumber")
            ).toThrow();
            expect(fn("tagName", "DINT", "A Description", true, null, 12)).toThrow();
            expect(
                fn("tagName", "DINT", "A Description", true, "aTarget", 12)
            ).not.toThrow();
        });

        it("Initializes with Desired Document Model", () => {
            let mem = new Member("tagName", "DINT");
            expect(mem).toMatchSnapshot();

            mem = new Member("tagName", "BOOL", null, false, "aTarget", 0);
            expect(mem).toMatchSnapshot();

            mem = new Member("tagName", "BIT", null, false, "aTarget", 0);
            expect(mem).toMatchSnapshot();

            mem = new Member("tagName", "DINT", "A Description");
            expect(mem).toMatchSnapshot();

            mem = new Member("tagName", "DINT", "A Description", true);
            expect(mem).toMatchSnapshot();

            mem = new Member("tagName", "DINT", "A Description", false, "ATarget");
            expect(mem).toMatchSnapshot();

            mem = new Member("tagName", "DINT", "A Description", false, "ATarget", 12);
            expect(mem).toMatchSnapshot();
        });
    });

    describe("Static Methods", () => {
        test("isMember: Returns Appropriate Judgement", () => {
            let mem = new Member("tagName", "DINT");
            expect(Member.isMember(mem)).toBeTruthy();

            mem = { notATag: 12 };
            expect(Member.isMember(mem)).toBeFalsy();

            mem = 12;
            expect(Member.isMember(mem)).toBeFalsy();
        });
    });
});
