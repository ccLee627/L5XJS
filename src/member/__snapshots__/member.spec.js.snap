// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Class New Instance Initializes with Desired Document Model 1`] = `
Member {
  "_dom": Object {
    "attributes": Object {
      "DataType": "DINT",
      "Hidden": false,
      "Name": "tagName",
      "Radix": "decimal",
    },
    "elements": Array [],
    "name": "Member",
    "type": "element",
  },
  "class_id": "cdb6dcc14d327d3c45e6e77e6648d07e",
  "elem_id": "33cb4159377b63ce9db11667160e39df",
}
`;

exports[`Member Class New Instance Initializes with Desired Document Model 2`] = `
Member {
  "_dom": Object {
    "attributes": Object {
      "BitNumber": 0,
      "DataType": "BIT",
      "Hidden": false,
      "Name": "tagName",
      "Radix": "binary",
      "Target": "aTarget",
    },
    "elements": Array [],
    "name": "Member",
    "type": "element",
  },
  "class_id": "cdb6dcc14d327d3c45e6e77e6648d07e",
  "elem_id": "33cb4159377b63ce9db11667160e39df",
}
`;

exports[`Member Class New Instance Initializes with Desired Document Model 3`] = `
Member {
  "_dom": Object {
    "attributes": Object {
      "BitNumber": 0,
      "DataType": "BIT",
      "Hidden": false,
      "Name": "tagName",
      "Radix": "binary",
      "Target": "aTarget",
    },
    "elements": Array [],
    "name": "Member",
    "type": "element",
  },
  "class_id": "cdb6dcc14d327d3c45e6e77e6648d07e",
  "elem_id": "33cb4159377b63ce9db11667160e39df",
}
`;

exports[`Member Class New Instance Initializes with Desired Document Model 4`] = `
Member {
  "_dom": Object {
    "attributes": Object {
      "DataType": "DINT",
      "Hidden": false,
      "Name": "tagName",
      "Radix": "decimal",
    },
    "elements": Array [
      Object {
        "elements": Array [
          Object {
            "cdata": "A Description",
            "type": "cdata",
          },
        ],
        "name": "Description",
        "type": "element",
      },
    ],
    "name": "Member",
    "type": "element",
  },
  "class_id": "cdb6dcc14d327d3c45e6e77e6648d07e",
  "elem_id": "33cb4159377b63ce9db11667160e39df",
}
`;

exports[`Member Class New Instance Initializes with Desired Document Model 5`] = `
Member {
  "_dom": Object {
    "attributes": Object {
      "DataType": "DINT",
      "Hidden": true,
      "Name": "tagName",
      "Radix": "decimal",
    },
    "elements": Array [
      Object {
        "elements": Array [
          Object {
            "cdata": "A Description",
            "type": "cdata",
          },
        ],
        "name": "Description",
        "type": "element",
      },
    ],
    "name": "Member",
    "type": "element",
  },
  "class_id": "cdb6dcc14d327d3c45e6e77e6648d07e",
  "elem_id": "33cb4159377b63ce9db11667160e39df",
}
`;

exports[`Member Class New Instance Initializes with Desired Document Model 6`] = `
Member {
  "_dom": Object {
    "attributes": Object {
      "DataType": "DINT",
      "Hidden": false,
      "Name": "tagName",
      "Radix": "decimal",
    },
    "elements": Array [
      Object {
        "elements": Array [
          Object {
            "cdata": "A Description",
            "type": "cdata",
          },
        ],
        "name": "Description",
        "type": "element",
      },
    ],
    "name": "Member",
    "type": "element",
  },
  "class_id": "cdb6dcc14d327d3c45e6e77e6648d07e",
  "elem_id": "33cb4159377b63ce9db11667160e39df",
}
`;

exports[`Member Class New Instance Initializes with Desired Document Model 7`] = `
Member {
  "_dom": Object {
    "attributes": Object {
      "DataType": "DINT",
      "Hidden": false,
      "Name": "tagName",
      "Radix": "decimal",
    },
    "elements": Array [
      Object {
        "elements": Array [
          Object {
            "cdata": "A Description",
            "type": "cdata",
          },
        ],
        "name": "Description",
        "type": "element",
      },
    ],
    "name": "Member",
    "type": "element",
  },
  "class_id": "cdb6dcc14d327d3c45e6e77e6648d07e",
  "elem_id": "33cb4159377b63ce9db11667160e39df",
}
`;
